symengine/functions.cpp

```diff
@@ -165,8 +165,7 @@ bool could_extract_minus(const Basic &arg)
     if (is_a_Number(arg)) {
         if (down_cast<const Number &>(arg).is_negative()) {
             return true;
-            // TODO: see #915
-        } else if (is_a<Complex>(arg)) {
+        } else if (is_a_Complex(arg)) {
             const Complex &c = down_cast<const Complex &>(arg);
```

Comment:
Ah! I guess implementing `get_real()` and `get_imag()` in `ComplexBase` should fix that. Currently, only class `Complex` doen't have these functions. Then we won't need to down_cast(). Correct me if I'm wrong?